### PR TITLE
[Caching] Add a cached label for cached pods

### PR DESCRIPTION
**Pull request by Bobgy**
/assign @Ark-kun 
I think we should still have the label in place. We'll very likely need it later.

### CODE REVIEW

File: backend/src/cache/server/mutation.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ import (
 const (
 	KFPCacheEnabledLabelKey   string = "pipelines.kubeflow.org/cache_enabled"
 	KFPCacheEnabledLabelValue string = "true"
+	KFPCachedLabelKey         string = "pipelines.kubeflow.org/cached"
```
Comment (Ark-kun) [inline backend/src/cache/server/mutation.go:37]: What do you think about "pipelines.kubeflow.org/reused_from_cache"?

Comment (Bobgy) [reply backend/src/cache/server/mutation.go]: Sure, sounds better!

**Pull request merged**